recipe-client-addon/lib/Utils.jsm

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+this.EXPORTED_SYMBOLS = ["Utils"];
+
+this.Utils = {
+  /**
+   * Convert an array of objects to an object. Each item is keyed by the value
+   * of the given key on the item.
+   *
+   * > list = [{foo: "bar"}, {foo: "baz"}]
+   * > keyBy(list, "foo") == {bar: {foo: "bar"}, baz: {foo: "baz"}}
+   *
+   * @param  {Array} list
+   * @param  {String} key
+   * @return {Object}
+   */
+  keyBy(list, key) {
+    return list.reduce((map, item) => {
+      if (!(key in item)) {
+        throw new Error(`List item was missing key "${key}".`);
```

Comment:
It's caught when running filter expressions, and not when fetching actions. Thinking more, I think I want this to log a warning and skip instead of throwing. This PR adds graceful handling of missing actions, which is where this wouldn't catch previously, and I think running other actions if only one of them is malformed from the server is better.